design/baremetal-operator/node-specific-kernel-args.md

```diff
@@ -0,0 +1,201 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# pre-provisioning kernel argument support
+
+## Status
+
+planned
+
+## Summary
+
+This document proposes a new BareMetalHost spec attribute intended for
+configuring the pre-provisioning image's (usually IPA) kernel arguments.
+
+## Motivation
+
+There are situations where a user would need node specific kernel arguments
+for the pre-provisioning image. The reason could be the use of a mixed set of
+hardware, troubleshooting provisioning or boot issues, selectively configuring
+custom logic e.g. enabling IPA plugins or other services embedded in the
+pre-provisioning image.
+
+### Goals
+
+1. Implement a string type optional spec field named `preprovisioningExtraKernelParams`
+2. Extend the generic `Provisioner` interface to support the new optional field
+3. Implement support for the `preprovisioningExtraKernelParams` in the Ironic provisioner
+4. Combine the values of the `preprovisioningExtraKernelParams` and the
+   `preprovisioningImage`'s `status.extraKernelParams` field if a
+   `preprovisioningImage` is refferences by the BMH and teh `status.extraKernelParams`
+   is not empty
+5. Introducing a `preprovisioningExtraKernelParams` related status field that
+   would display the last applied `preprovisioningExtraKernelParams` value.
+   Let's call this field `status.lastAppliedPreprovisioningExtraKernelParams`
+
+### Non-Goals
+
+1. To implement any sort validation for the new optional field
+2. Providing a default value for the new spec field
+3. Implement conditional logic to overwrite default kernel parameters on
+   per node basis
+
+## Proposal
+
+This document proposes a new BareMetalHost spec attribute intended for
+configuring the pre-provisioning image's (usually IPA) kernel arguments.
+
+The proposal involves the introduction of a new spec field thus a modification
+to the BMO API. The API change is intended to be backward compatible as the
+new `preprovisioningExtraKernelParams` field is optional.
+
+### Implementation Details/Notes/Constraints
+
+The `preprovisioningExtraKernelParams` field would be located directly under `spec`
+section. The format of the new field would be a list/array of individual
```

Comment:
This has been changes elsewhere to be a string

```suggestion
section. The format of the new field would be a string.
The new field would be optional and it would be considered
```